draft-ietf-quic-transport.md

```diff
@@ -4722,6 +4719,37 @@ receiver, an off-path attacker will need to race the duplicate packet against
 the original to be successful in this attack.  Therefore, QUIC receivers ignore
 ECN codepoints set in duplicate packets (see {{using-ecn}}).
 
+## Stateless Reset Oracle {#reset-oracle}
+
+An attacker that can cause a server to emit a stateless reset token can cause
+any connection that uses the same connection ID and method of producing the
+token to be destroyed.  This creates a possible denial of service attack on
+existing connections if the attacker is able to cause a stateless reset token to
+be generated for a connection with a selected connection ID.  That is, an attack
+is possible if an attacker can alter packet routing so that a packet is received
+by an instance that uses the same stateless reset key, but no connection state
+for the connection ID.
+
+The only viable defense against this style of denial of service is to ensure
+that routing configuration for a cluster guarantees consistent routing of
+packets for a given connection ID.  That is, the routing needs to ensure that a
+stateless reset is not generated unless the associated connection (as identified
+by the connection ID) cannot proceed.
+
+For endpoints that operate in clusters, instances that share a static key for
+stateless reset (see {{reset-token}}) MUST be arranged so that packets with a
+given connection ID always arrive at an instance that has connection state,
+unless that connection is no longer active.
+
+In the case of a cluster that uses dynamic load balancing, it's possible that a
+change in load balancer configuration could happen while an active instance
+retains connection state; even if an instance retains connection state, the
+change in routing and resulting stateless reset will result in the connection
+being terminated.  This is acceptable as long as the loss of connections during
+rebalancing is by design and not something that an external entity can
+influence.
```

Comment:
I am not sure how the last sentence applies to the discussion in the paragraph.

If I understand correctly, the intent of the paragraph is to state that if there is no chance in the packet being routed to the correct server, it is logical to send a stateless reset rather than waiting for timeout. That is logical. However, the last sentence talks about frequency of rebalancing, which IMO is unrelated to whether if we send a reset or wait for timeout.

It seems to me that it might be a better idea to state here that, by using the rebalancing as an example, that sending stateless reset is preferable than waiting for timeout, and move the discussion about the frequency of rebalancing to somewhere else, since it is not an issue specific to QUIC having authenticated resets (TCP sharding using 5-tuple has the same issue).